nncf/sparsity/schedulers.py

```diff
@@ -236,3 +236,52 @@ def current_sparsity_level(self):
     @property
     def target_sparsity_level(self):
         return self.sparsity_levels[-1]
+
+
+@SPARSITY_SCHEDULERS.register("per_layer")
+class PerLayerSparsityScheduler(SparsityScheduler):
+    def __init__(self, sparsity_algo, params, sparsity_level_control_algo=None):
+        super().__init__(sparsity_algo, params)
+        self.sparsified_modules_info_by_modules_name = {}
+        self.per_layer_sparsity_levels = {}
+        for minfo in self.algo.sparsified_module_info:
+            self.sparsified_modules_info_by_modules_name[minfo.module_name] = minfo
+        self.layered_sparsity_level_control_algo = sparsity_level_control_algo
```

Comment:
You don't need to register the control algorithm in the scheduler at all. It is the control algorithm that owns and uses the scheduler according to the scheduler's API, not vice versa.

Reply:
Deleted.